src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,674 @@
+    FuzzingData(const uint8_t* data, size_t size) : data(data), size(size - UtilityDataSize), Ptrs(), NumIterated(), Cyclic(true) {}
+    bool Initialize() {
+        Ptrs = 0;
+        NumIterated = 0;
+        return true;
+    }
+    template<typename T>
+    bool TryGetRandom(T UpperBound, T* Val) {
+        int type_size = sizeof(T);
+        if (!CheckBoundary(type_size)) {
+            return false;
+        }
+        memcpy(Val, &data[Ptrs], type_size);
+        *Val = (T)(*Val % UpperBound);
+        Ptrs += type_size;
+        return true;
+    }
+    size_t GetIterateCount() {
+        return NumIterated;
+    }
+};
+
+static FuzzingData* FuzzData = nullptr;
+
+template<typename T>
+T GetRandom(T UpperBound) {
+    if (!FuzzData) {
+        return (T)(rand() % (int)UpperBound);
+    }
+    uint64_t out = 0;
+
+    if ((uint64_t)UpperBound <= 0xff) {
+        (void)FuzzData->TryGetRandom((uint8_t)UpperBound, (uint8_t*)&out);
+    } else if ((uint64_t)UpperBound <= 0xffff) {
+        (void)FuzzData->TryGetRandom((uint16_t)UpperBound, (uint16_t*)&out);
+    } else if ((uint64_t)UpperBound <= 0xffffffff) {
+        (void)FuzzData->TryGetRandom((uint32_t)UpperBound, (uint32_t*)&out);
+    } else {
+        (void)FuzzData->TryGetRandom((uint64_t)UpperBound, &out);
+    }
+    return (T)out;
+}
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+_Function_class_(CXPLAT_DATAPATH_RECEIVE_CALLBACK)
+void
+UdpRecvCallback( 
+    _In_ CXPLAT_SOCKET* /* Binding */,
+    _In_ void* /* Context */,
+    _In_ CXPLAT_RECV_DATA* RecvBufferChain
+    )
+{
+    CxPlatRecvDataReturn(RecvBufferChain);
+}
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+_Function_class_(CXPLAT_DATAPATH_UNREACHABLE_CALLBACK)
+void
+UdpUnreachCallback(
+    _In_ CXPLAT_SOCKET* /* Binding */,
+    _In_ void* /* Context */,
+    _In_ const QUIC_ADDR* /* RemoteAddress */
+    )
+{
+}
+
+void printf_buf(const char* name, void* buf, uint32_t len)
+{
+    printf("%s: ", name);
+    for (uint32_t i = 0; i < len; i++) {
+        printf("%.2X", ((uint8_t*)buf)[i]);
+    }
+    printf("\n");
+}
+
+struct TlsContext
+{
+    CXPLAT_TLS* Ptr;
+    CXPLAT_SEC_CONFIG* SecConfig;
+    CXPLAT_TLS_PROCESS_STATE State;
+    uint8_t AlpnListBuffer[256];
+
+    TlsContext() :
+        Ptr(nullptr), SecConfig(nullptr) {
+
+        AlpnListBuffer[0] = (uint8_t)strlen(Alpn);
+        memcpy(&AlpnListBuffer[1], Alpn, AlpnListBuffer[0]);
+
+        CxPlatZeroMemory(&State, sizeof(State));
+        State.Buffer = (uint8_t*)CXPLAT_ALLOC_NONPAGED(8000, QUIC_POOL_TOOL);
+        State.BufferAllocLength = 8000;
+
+        QUIC_CREDENTIAL_CONFIG CredConfig = {
+            QUIC_CREDENTIAL_TYPE_NONE,
+            QUIC_CREDENTIAL_FLAG_CLIENT | QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION,
+            NULL, NULL, NULL, NULL
+        };
+        CXPLAT_TLS_CALLBACKS TlsCallbacks = {
+            OnRecvQuicTP,
+            NULL
+        };
+        QUIC_SUCCEEDED(
```

Comment:
There's no `if` statement here. So either add one, or remove the `QUIC_SUCCEEDED`.